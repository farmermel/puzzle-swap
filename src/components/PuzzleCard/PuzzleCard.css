.puzzle-card {
  display: inline-block;
  border-radius: 6px;
  background-color: #fdfdfd;
  box-shadow: 1px 1px 2px #64564C;
  margin: 10px;
  height: 480px;
  width: 300px;
}

.puzzle-card h3 {
  font-size: 1.7em;
}

.puzzle-img {
  width: 100%;
  height: 210px;
  /*background-size: contain;*/
  background-repeat: no-repeat;
  background-size: cover;
}