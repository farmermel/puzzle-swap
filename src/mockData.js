export const mockPuzzles = [
  {
    title: 'Witch',
    numPieces: '1000',
    piecesMissing: '3-4',
    imgUrl: 'google',
    puzzleId: 4
  },
  {
    title: 'Fairy',
    numPieces: '1000',
    piecesMissing: '3-4',
    imgUrl: 'yahoo',
    puzzleId: 45
  },
  {
    title: 'Unicorn',
    numPieces: '1500',
    piecesMissing: '3-4',
    imgUrl: 'aol',
    puzzleId: 6
  }
]