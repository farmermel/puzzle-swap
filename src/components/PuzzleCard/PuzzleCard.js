import React from 'react';
import PropTypes from 'prop-types';
import './PuzzleCard.css';

const determineDisable = (handleClaim, handleDelete, puzzleId, ownerId, userId) => {
  return ownerId === userId 
    ? renderDelete(handleDelete, puzzleId) 
    : renderClaim(handleClaim, ownerId);
}

const renderDelete = (handleDelete, puzzleId) => {
  return (
    <button onClick={() => handleDelete(puzzleId)}
            className='claim'>Unlist</button>
  )
}

const renderClaim = (handleClaim, ownerId) => {
  return (
    <button onClick={() => handleClaim(ownerId)}
            className='claim'>Claim</button>
  )
}

const PuzzleCard = ({ handleClaim, handleDelete, puzzle, user }) => {
  const inlineStyles = {
    backgroundImage: `url(${puzzle.imgUrl})`
  }
  const { puzzleId, userId: ownerId } = puzzle;
  return (
    <article className='puzzle-card'>
      <div className='puzzle-img' style={inlineStyles}></div>
      <div className='puzzle-txt-wrap'>
        <h3>{puzzle.title}</h3>
        <p>Pieces: {puzzle.numPieces}</p>
        <p>Pieces missing: {puzzle.piecesMissing}</p>
        {
          user 
            ? determineDisable(handleClaim, handleDelete, puzzleId, ownerId, user.uid)
            : <button>Sign in to claim</button>
        }
      </div>
    </article>
  )
}

PuzzleCard.propTypes = {
  puzzle: PropTypes.shape({
    title: PropTypes.string,
    numPieces: PropTypes.string,
    piecesMissing: PropTypes.string,
    imgUrl: PropTypes.string,
    puzzleId: PropTypes.number
  }),
  handleClaim: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired
}

export default PuzzleCard;