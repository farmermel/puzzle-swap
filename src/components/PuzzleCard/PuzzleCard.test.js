import React from 'react';
import { shallow } from 'enzyme';
import PuzzleCard from './PuzzleCard';
import { mockPuzzles } from '../../mockData';

describe('PuzzleCard', () => {
  const mockPuzzle = mockPuzzles[0]
  it('should exist', () => {
    const wrapper = shallow(<PuzzleCard puzzle={ mockPuzzle } />);
    expect(wrapper).toBeDefined();
  })
})