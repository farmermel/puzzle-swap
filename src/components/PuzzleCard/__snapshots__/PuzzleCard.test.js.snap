// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PuzzleCard matches snapshot if user is logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PuzzleCard
    handleClaim={[Function]}
    puzzle={
        Object {
            "imgUrl": "google",
            "numPieces": "1000",
            "piecesMissing": "3-4",
            "puzzleId": 4,
            "title": "Witch",
          }
    }
    user={
        Object {
            "uid": "5",
            "username": "Nymeria",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="puzzle-img"
          style={
                    Object {
                              "backgroundImage": "url(google)",
                            }
          }
/>,
        <h3>
          Witch
</h3>,
        <p>
          Pieces: 
          1000
</p>,
        <p>
          Pieces missing: 
          3-4
</p>,
        <button
          className="claim"
          disabled={false}
          onClick={[Function]}
>
          Claim
</button>,
      ],
      "className": "puzzle-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "puzzle-img",
          "style": Object {
            "backgroundImage": "url(google)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Witch",
        },
        "ref": null,
        "rendered": "Witch",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Pieces: ",
            "1000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Pieces: ",
          "1000",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Pieces missing: ",
            "3-4",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Pieces missing: ",
          "3-4",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Claim",
          "className": "claim",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Claim",
        "type": "button",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="puzzle-img"
            style={
                        Object {
                                    "backgroundImage": "url(google)",
                                  }
            }
/>,
          <h3>
            Witch
</h3>,
          <p>
            Pieces: 
            1000
</p>,
          <p>
            Pieces missing: 
            3-4
</p>,
          <button
            className="claim"
            disabled={false}
            onClick={[Function]}
>
            Claim
</button>,
        ],
        "className": "puzzle-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "puzzle-img",
            "style": Object {
              "backgroundImage": "url(google)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Witch",
          },
          "ref": null,
          "rendered": "Witch",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Pieces: ",
              "1000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Pieces: ",
            "1000",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Pieces missing: ",
              "3-4",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Pieces missing: ",
            "3-4",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Claim",
            "className": "claim",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Claim",
          "type": "button",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PuzzleCard matches snapshot if user is not logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PuzzleCard
    handleClaim={[Function]}
    puzzle={
        Object {
            "imgUrl": "google",
            "numPieces": "1000",
            "piecesMissing": "3-4",
            "puzzleId": 4,
            "title": "Witch",
          }
    }
    user={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="puzzle-img"
          style={
                    Object {
                              "backgroundImage": "url(google)",
                            }
          }
/>,
        <h3>
          Witch
</h3>,
        <p>
          Pieces: 
          1000
</p>,
        <p>
          Pieces missing: 
          3-4
</p>,
        <h3>
          Sign in to claim
</h3>,
      ],
      "className": "puzzle-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "puzzle-img",
          "style": Object {
            "backgroundImage": "url(google)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Witch",
        },
        "ref": null,
        "rendered": "Witch",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Pieces: ",
            "1000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Pieces: ",
          "1000",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Pieces missing: ",
            "3-4",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Pieces missing: ",
          "3-4",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign in to claim",
        },
        "ref": null,
        "rendered": "Sign in to claim",
        "type": "h3",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="puzzle-img"
            style={
                        Object {
                                    "backgroundImage": "url(google)",
                                  }
            }
/>,
          <h3>
            Witch
</h3>,
          <p>
            Pieces: 
            1000
</p>,
          <p>
            Pieces missing: 
            3-4
</p>,
          <h3>
            Sign in to claim
</h3>,
        ],
        "className": "puzzle-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "puzzle-img",
            "style": Object {
              "backgroundImage": "url(google)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Witch",
          },
          "ref": null,
          "rendered": "Witch",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Pieces: ",
              "1000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Pieces: ",
            "1000",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Pieces missing: ",
              "3-4",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Pieces missing: ",
            "3-4",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign in to claim",
          },
          "ref": null,
          "rendered": "Sign in to claim",
          "type": "h3",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
